server/game/cards/11.2-TMoW/ServeObeyProtect.js

```diff
@@ -0,0 +1,29 @@
+const DrawCard = require('../../drawcard.js');
+
+class ServeObeyProtect extends DrawCard {
+    setupCardAbilities(ability) {
+        this.reaction({
+            when: {
+                onChallengeInitiated: event => event.challenge.defendingPlayer === this.controller &&
+                                               this.controller.getNumberOfChallengesLost('defender') >= 1
+            },
+            cost: ability.costs.returnToHand(card => card.isFaction('martell') && card.getType() === 'character'),
+            target: {
+                cardCondition: (card, context) => card.location === 'hand' && card.controller === this.controller &&
+                                                  card.isFaction('martell') && card.getType() === 'character' &&
+                                                  card.getPrintedCost() <= 5 && 
+                                                  (!context.costs.returnToHand || card !== context.costs.returnToHand)
```

Comment:
Should "different character" mean a card with a different name, rather than a card other than the one returned?

Reply:
Not sure yet, it's been asked to the designers.

We've never had a true definition of different, and there's a bunch of other cards it impacts as well, like "Steel Rain" and the jumping Direwolf attachments like Lady.

Reply:
k, will merge for now and we can revisit when there's an official ruling.